app/repositories/UserRepository.php

```diff
@@ -228,24 +228,39 @@ public function searchUsers($query) {
         $stmt->execute(['query' => '%' . $query . '%']);
         return $stmt->fetchAll(PDO::FETCH_ASSOC); // Return users as an associative array
     }
-    public function searchAllUsers($query) {
-        $stmt = $this->db->prepare("
-            SELECT u.user_id, u.username, u.email, 
-                   u.account_status_id,
+    public function searchAllUsers($query = null) {
+        $sql = "
+            SELECT u.user_id, u.username, u.email, u.account_status_id,
                    CASE 
                        WHEN u.account_status_id = 1 THEN 'Active'
                        WHEN u.account_status_id = 2 THEN 'Suspended'
                        WHEN u.account_status_id = 3 THEN 'Banned'
                        ELSE 'Unknown'
-                   END AS account_status_name
+                   END AS account_status_name,
+                   COALESCE(p.profile_picture, 'user-default.svg') AS profile_picture
             FROM users u
-            WHERE u.username LIKE :query
-        ");
-        $stmt->execute(['query' => '%' . $query . '%']);
+            LEFT JOIN user_profile p ON u.user_id = p.user_id
```

Comment:
This function looks longer than it needs to be, is there a reason for that ?

Reply:
We wanted to implement javascript for some of the related functionality, this means some extra lifting from the php end, in this case the query used in the repository for this function.

This function is designed to fetch user data with optional filtering, handling account status labels and profile pictures in one query:

SQL Query: Selects user details, with:

CASE for readable account status ("Active," "Suspended," etc.).
COALESCE for default profile picture if missing.
LEFT JOIN to include profile picture data without excluding users without one.
Query Filtering: If a query (username search) is provided, a WHERE clause filters results to match the search term.